src/compiler/data.ts

```diff
@@ -82,18 +86,18 @@ data.source.formatParse = function(encoding) {
  * Generate Vega transforms for the source data table.  This can include
  * transforms for time unit, binning and filtering.
  */
-data.source.transform = function(encoding) {
+export function transform(encoding) {
   // null filter comes first so transforms are not performed on null values
   // time and bin should come before filter so we can filter by time and bin
-  return data.source.transform.nullFilter(encoding).concat(
-    data.source.transform.formula(encoding),
-    data.source.transform.time(encoding),
-    data.source.transform.bin(encoding),
-    data.source.transform.filter(encoding)
+  return nullFilterTransform(encoding).concat(
```

Comment:
I think there should be a better way to do nesting. For example a module within a module.